lib/hanami/components/components.rb

```diff
@@ -177,9 +177,11 @@ module Components # rubocop:disable Metrics/ModuleLength
       end
 
       resolve do |configuration|
-        Hanami::Mailer.configuration = Hanami::Mailer::Configuration.new if Hanami.code_reloading?
-        Hanami::Mailer.configure(&configuration.mailer)
-        Hanami::Mailer.configuration
+        if configuration.mailer
```

Comment:
The trouble is that `mailer` is a hard dependency of `hanami`, unlike `model`.

If we ignore the configuration silently we may happen into issues complaining about it not working at all.

Either we make it a soft-dependency or put in a warn that strange stuff may happen.